### ISSUE TITLE
Why the  bbox_pred need to multiply with the  stride?

### ISSUE THREAD

**Issue by gyftsy** [strongwolf/DW#12]
hi @strongwolf 
I have some problems with this part of the code.
If I understand correctly, the regressor of the fcos network is the distance from the center grid point to the four sides. This distance is the distance on the original image. You did not introduce stride information, including the use of distance to decode box in loss. But you're multiplying the regressor by the step size on line 7. So I don't understand why the regressor is multiplied by the stride here. In this case, decoded_bbox_pred is right? Or the deform_sampling need to change in this way?

`def forward_single(self, x, scale, stride):
        b, c, h, w = x.shape
        cls_score, bbox_pred, cls_feat, reg_feat = super().forward_single(x)
        centerness = self.conv_centerness(reg_feat)
        bbox_pred = scale(bbox_pred).float()
        bbox_pred = F.relu(bbox_pred)
        bbox_pred *= stride
        if self.with_reg_refine:
            reg_dist = bbox_pred.permute(0, 2, 3, 1).reshape(-1, 4)
            points = self.prior_generator.single_level_grid_priors((h,w), self.strides.index(stride), dtype=x.dtype, device=x.device)
            points = points.repeat(b, 1) 
            decoded_bbox_preds = distance2bbox(points, reg_dist).reshape(b, h, w, 4).permute(0, 3, 1, 2)
            reg_offset = self.reg_offset(reg_feat)
            bbox_pred_d  = bbox_pred / stride 
            reg_offset = torch.stack([reg_offset[:,0], reg_offset[:,1] - bbox_pred_d[:, 0],\
                                        reg_offset[:,2] - bbox_pred_d[:, 1], reg_offset[:,3],
                                        reg_offset[:,4], reg_offset[:,5] + bbox_pred_d[:, 2],
                                        reg_offset[:,6] + bbox_pred_d[:, 3], reg_offset[:,7],], 1)
            bbox_pred = self.deform_sampling(decoded_bbox_preds.contiguous(), reg_offset.contiguous()) 
            bbox_pred = F.relu(bbox2distance(points, bbox_pred.permute(0, 2, 3, 1).reshape(-1, 4)).reshape(b, h, w, 4).permute(0, 3, 1, 2).contiguous())
        
        return cls_score, bbox_pred, centerness`

**Comment by gyftsy** [comment]
Can you give me your email or contact information if possible? I am integrating dw into yolox, but have encountered some problems, I hope to get your help, thank you very much.

**Comment by gyftsy** [comment]
Problem has been resolved. After migrating dw to yolox, the accuracy drops by 3-4 maps.

**Comment by strongwolf** [comment]
Obviously, there is no detector that directly regresses the absolute distance in the original image space.  We will try to implement a standard DW into yolox soon. Thanks
